sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -57,8 +57,60 @@ static int max98373_hw_params(struct snd_pcm_substream *substream,
 	return 0;
 }
 
+static int max98373_trigger(struct snd_pcm_substream *substream, int cmd)
+{
+	struct snd_soc_pcm_runtime *rtd = substream->private_data;
+	struct snd_soc_dai *codec_dai;
+	int j, ret;
+
+	for_each_rtd_codec_dais(rtd, j, codec_dai) {
+		const char *name = codec_dai->component->name;
+		struct snd_soc_component *component = codec_dai->component;
+		struct snd_soc_dapm_context *dapm =
+				snd_soc_component_get_dapm(component);
+		char pin_name[20];
+
+		if (strcmp(name, MAX_98373_DEV0_NAME) &&
+			strcmp(name, MAX_98373_DEV1_NAME))
```

Comment:
Is this check really needed? I mean the trigger op is only set for the max_98373 BE DAI link, right?

Reply:
removed